tests/hypothesis_strategies/acquisition.py

```diff
@@ -0,0 +1,42 @@
+"""Hypothesis strategies for acquisition functions."""
+
+import hypothesis.strategies as st
+
+from baybe.acquisition import (
+    ExpectedImprovement,
+    ProbabilityOfImprovement,
+    UpperConfidenceBound,
+    qExpectedImprovement,
+    qProbabilityOfImprovement,
+    qUpperConfidenceBound,
+)
+from baybe.acquisition.base import AcquisitionFunction
+from baybe.utils.basic import get_subclasses
+
+acqf_long_names = [cl.__name__ for cl in get_subclasses(AcquisitionFunction)]
+acqf_abbreviations = [
+    cl._abbreviation
+    for cl in get_subclasses(AcquisitionFunction)
+    if hasattr(cl, "_abbreviation")
+]
+
+acqf_objects = st.one_of(
+    st.sampled_from(
```

Comment:
The sampling here is unbalanced

Reply:
can you elaborate? Ive put the more defaulty-seeming things in the front, as suggested by hypothesis, but I dont know what you mean with your comment

given that I will remove the other non-object parts this `acqf_objects` will become the new `random_acqfs`

Reply:
Putting simple stuff first is good. But you need to also remove the `sampled_from` layer (why did you introduce it in the first place?). Otherwise the sampling goes like "draw from the outer layer first, then from the inner layer" which skews your distribution to the bottom part of the list

Reply:
for me it was just a shortcut to not have so many `st.builds` calls but will change if thats an issue. Since the inner part are the simple functions without additional parameter, wouldnt a sampling with more entries from the more complex ones (with `beta`) even make sense?

Reply:
Ah, I see. So given that I'm also not a hypothesis expert, there is one golden rule I have learned so far, which is "don't mess with the random generation, hypothesis will do the job for you". That said, please note that my original "unbalanced" comment was just a simplification because I didn't want to go into details. You have to keep in mind that what really happens behind the scenes is not uniform sampling but hypothesis will automatically search for edge cases and apply shrinkage. So the frequency with which stuff is sampled is anyway not controlled by us. With that in mind, I think the best approach is to give hypothesis the least complex generation path to start with. But this is only my opinion based on my (limited) experience with the framework so far...

Reply:
ok I get what you mean now, leaving the random selection to hypothesis imo makes sense and I wont introduce bias by my specification here

so changing it to several `st.builds` now, is that what you intend?

Reply:
pushed

Reply:
Yes, I think that looks good. Next improvement would be to actually expose these individual strategies also in the namespace, like I did for the other classes (because sometimes you just want to generate a specific subclass and I think it's good to have them organized in one place) but you can skip it in this PR. We can do so in that one overhaul where we align the constructor-vs-strategy thing...